src/crypto/hash.rs

```diff
@@ -112,6 +140,12 @@ impl DerefMut for Hash {
   }
 }
 
+impl fmt::Display for Hash {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    write!(f, "{:x?}", self.hash)
```

Comment:
for compat with the JS version we should probably use `pretty-fmt` here.

Reply:
Right, I thought there was somewhere I'd missed that!

Related question: are we aiming for full storage API compatibility with the js version, as well as network API compatibility? If so are the JS maintainers aware that any braking change they make will also break the rust version?